server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/routes/UserRoutes.java

```diff
@@ -224,6 +255,84 @@ private String verifyUser(Request request, Response response) throws UsersReposi
         }
     }
 
+    private List<String> getAuthorizedUsers(Request request, Response response) throws UsersRepositoryException {
+        Username baseUser = extractUsername(request);
+
+        if (userService.userExists(baseUser)) {
+            return Flux.from(delegationStore.authorizedUsers(baseUser))
+                .map(Username::asString)
+                .collectList()
+                .block();
+        } else {
+            throw ErrorResponder.builder()
+                .statusCode(HttpStatus.NOT_FOUND_404)
+                .type(ErrorType.INVALID_ARGUMENT)
+                .message(String.format("User '%s' does not exist", baseUser.asString()))
+                .haltError();
+        }
+    }
+
+    private String clearAllAuthorizedUsers(Request request, Response response) throws UsersRepositoryException {
+        Username baseUser = extractUsername(request);
+
+        if (userService.userExists(baseUser)) {
+            Mono.from(delegationStore.clear(baseUser)).block();
+            return Constants.EMPTY_BODY;
+        } else {
+            throw ErrorResponder.builder()
+                .statusCode(HttpStatus.NOT_FOUND_404)
+                .type(ErrorType.INVALID_ARGUMENT)
+                .message(String.format("User '%s' does not exist", baseUser.asString()))
+                .haltError();
+        }
+    }
+
+    private String addAuthorizedUser(Request request, Response response) throws UsersRepositoryException {
+        Username baseUser = extractUsername(request);
+        Username delegatedUser = extractDelegatedUsername(request);
+
+        if (!userService.userExists(baseUser)) {
+            throw ErrorResponder.builder()
+                .statusCode(HttpStatus.NOT_FOUND_404)
+                .type(ErrorType.INVALID_ARGUMENT)
+                .message(String.format("User '%s' does not exist", baseUser.asString()))
+                .haltError();
+        } else if (!userService.userExists(delegatedUser)) {
+            response.status(HttpStatus.CONFLICT_409);
```

Comment:
`409` status for this case? I think it should be 404. 
But I don't sure if exactly is 409, new to me

Reply:
rf: https://github.com/linagora/james-project/issues/4546#issue-1236728350

Reply:
409 -> conflict.

IMO thinking more about it it's more of a 400 (invalid) as the provided payload is invalid...